apps/src/code-studio/pd/workshop_dashboard/components/survey_results_header.jsx

```diff
@@ -86,24 +87,54 @@ const SurveyResultsHeader = React.createClass({
     const courses = this.getCoursesForWorkshops();
 
     if (courses.length > 0) {
-      this.filterWorkshops(courses[0]);
+      if (this.props.preselectedWorkshopId) {
+        const preselectedWorkshop = _.find(this.props.workshops, workshop => {
+          return workshop.id === parseInt(this.props.preselectedWorkshopId);
+        });
+
+        if (preselectedWorkshop) {
+          this.filterWorkshops(preselectedWorkshop.course, preselectedWorkshop);
+        } else {
+          this.filterWorkshops(courses[0]);
+        }
+      } else {
+        this.filterWorkshops(courses[0]);
+      }
     }
   },
 
-  filterWorkshops(course) {
+  filterWorkshops(course, preselectedWorkshop = undefined) {
     let filteredWorkshops = _.filter(this.props.workshops, function (workshop) {
       return workshop.course === course;
     });
 
-    let firstWorkshopId = this.props.organizerView ? undefined : (filteredWorkshops[0] ? filteredWorkshops[0].id : undefined);
 
-    this.setState({
-      selectedCourse: course,
-      filteredWorkshops: filteredWorkshops,
-      selectedWorkshopId: firstWorkshopId,
+    filteredWorkshops = _.sortBy(filteredWorkshops, workshop => {
+      return this.getWorkshopFriendlyName(workshop);
     });
 
-    this.setSurveyPanel(course, firstWorkshopId, firstWorkshopId && this.getWorkshopFriendlyName(filteredWorkshops[0]));
+    _.reverse(filteredWorkshops);
+
+
+    if (preselectedWorkshop) {
+      this.setState({
+        selectedCourse: course,
+        filteredWorkshops: filteredWorkshops,
+        selectedWorkshopId: preselectedWorkshop.id
+      });
+
+      this.setSurveyPanel(course, preselectedWorkshop.id, this.getWorkshopFriendlyName(preselectedWorkshop));
+    } else {
+      let firstWorkshopId = this.props.organizerView ? undefined : (filteredWorkshops[0] ? filteredWorkshops[0].id : undefined);
+
+      this.setState({
+        selectedCourse: course,
+        filteredWorkshops: filteredWorkshops,
+        selectedWorkshopId: firstWorkshopId,
+      });
```

Comment:
This is also a bit hard to read. I'd prefer a single setState. It shouldn't need to know about preselectedWorkshop. That should only dictate which workshop is selected. Course and filteredWorkshops don't change, so it doesn't need to repeat.

The logic here to select the first workshop is a bit complex, but maybe something like:
```ES6
let selectedWorkshopId;
if (preselectedWorkshop) {
  selectedWorkshopId = preselectedWorkshop.id;
else {
  selectedWorkshopId = this.props.organizerView ? undefined : (filteredWorkshops[0] ? filteredWorkshops[0].id : undefined);
}

this.setState({
  selectedCourse: course,
  filteredWorkshops,
  selectedWorkshopId
});
```